usaspending_api/search/tests/integration/test_spending_by_geography.py

```diff
@@ -2223,3 +2334,103 @@ def test_spending_by_geo_program_activity(
 
     assert resp.status_code == status.HTTP_200_OK
     assert expected_response == resp.json().get("results"), "Unexpected or missing content!"
+
+
+spending_level_test_params = [
+    (
+        "awards",
+        [
+            {
+                "shape_code": "CAN",
+                "display_name": "Test Canada",
+                "aggregated_amount": 600.0,
+                "population": 200,
+                "per_capita": 3.0,
+            },
+            {
+                "shape_code": "USA",
+                "display_name": "Test United States",
+                "aggregated_amount": 1000.0,
+                "population": 500,
+                "per_capita": 2.0,
+            },
+        ],
+    ),
+    (
+        "subawards",
+        [
+            {
+                "shape_code": "CAN",
+                "display_name": "Test Canada",
+                "aggregated_amount": 400.0,
+                "population": 200,
+                "per_capita": 2.0,
+            },
+            {
+                "shape_code": "USA",
+                "display_name": "Test United States",
+                "aggregated_amount": 1500.0,
+                "population": 500,
+                "per_capita": 3.0,
+            },
+        ],
+    ),
+    (
+        "transactions",
+        [
+            {
+                "shape_code": "CAN",
+                "display_name": "Test Canada",
+                "aggregated_amount": 500.0,
+                "population": 200,
+                "per_capita": 2.5,
+            },
+            {
+                "shape_code": "USA",
+                "display_name": "Test United States",
+                "aggregated_amount": 500.0,
+                "population": 500,
+                "per_capita": 1.0,
+            },
+        ],
+    ),
+]
+
+
+@pytest.mark.parametrize("spending_level,expected_results", spending_level_test_params)
```

Comment:
Not worth blocking over, but what is the benefit in using this decorator versus defining these parameters in the test function below? Question pertains to readability of the tests if we followed this pattern for multiple tests down the road.

Reply:
I think the main benefit is that pytest runs each parameter as a separate test allowing you to more easily pinpoint failures:
```
usaspending_api/search/tests/integration/test_spending_by_geography.py::test_correct_response_with_spending_level[awards-expected_results0] PASSED                                                                                                                 [ 88%]
usaspending_api/search/tests/integration/test_spending_by_geography.py::test_correct_response_with_spending_level[subawards-expected_results1] PASSED                                                                                                              [ 94%]
usaspending_api/search/tests/integration/test_spending_by_geography.py::test_correct_response_with_spending_level[transactions-expected_results2] PASSED                                                                                                           [100%]
```